modules/affile/logproto-file-writer.c

```diff
@@ -153,16 +148,16 @@ log_proto_file_writer_flush(LogProtoClient *s)
   // emit a "flush" signal to the transport, so that it can react to the error
   GString *request_body = g_string_new("");
   g_string_append_printf(request_body, "%d", self->buf_size);
-  HttpHeaderRequestSignalData signal_data =
+  FileRotationRequestData signal_data =
   {
-    .result = HTTP_SLOT_SUCCESS,
-    .request_headers = self->worker->request_headers,
-    .request_body = request_body,
+    .filename = self->super.transport->name,
+    .size = self->buf_size, // implementation yet to be decided
+    .interval = 0, // implementation yet to be decided
   };
 
-  HTTPDestinationDriver *owner = (HTTPDestinationDriver *) self->worker->super.owner;
+  FileRotationDriver *owner = (FileRotationDriver *) self->owner;
 
-  EMIT(owner->super.super.super.super.signal_slot_connector, signal_http_header_request, &signal_data);
+  EMIT(owner->super.super.super.super.signal_slot_connector, signal_file_rotation_request, &signal_data);
```

Comment:
This probably does not work due to the above reasoning.

Reply:
So it should be http header request?

Reply:
No, it should be a `signal_file_rotation_request`, which needs to be declared in `file-signals.h`.

The problem here is that we need a `signal_slot_connector` instance, but not a completely new instance, we need `AFFileDestDriver`'s signal slot connector.
In order not to reference `AFFileDestDriver` directly from `LogProtoFileWriter` (unnecessary coupling), we could pass `AFFileDestDriver`'s signal slot connector to `log_proto_file_writer_new()` as a parameter.

Reply:
Fixed :+1: